examples/glue_benchmark/glue.py

```diff
@@ -115,12 +127,17 @@ def split_features(x):
             f"glue/{task_name}",
             split=["train", "test" + test_suffix, "validation" + test_suffix],
         )
+
+    # Extract out the index order of test dataset.
+    idx_order = get_test_data_idx_order(test_ds)
```

Comment:
I think this could just be `test_ds.map(lambda data: data["idx"])` right? That would be a little more readable.

Reply:
that would return a tf dataset, but that could actually be used directly below (and would simplify the return here), will comment below

Reply:
thanks for the suggestion! I prefer to convert it to a list because it is lighter than tf Dataset, especially for debugging purpose.

Reply:
It just seems a bit confusing that we return three datasets and one list from this function. It may be less obvious to a reader what you are doing than you think! Took me a second to grok this.

If you really would like a list here, I suggest the following:

```
idx_order = list(test_ds.map(lambda d: d["idx"]).as_numpy_iterator())
```

Reply:
Yea, if you find it confusing, I am happy to change the code.

Reply:
I would return all datasets from this function, and convert to a list later if you would like a list. `list(ds.as_numpy_iterator())` is the secret sauce for that. You could do that inside `generate_submission_files`.

But both OK, you should make the final call here!